custom_components/hacs/utils/queue_manager.py

```diff
@@ -18,26 +19,26 @@ class QueueManager:
 
     def __init__(self, hass: HomeAssistant) -> None:
         self.hass = hass
-        self.queue: list[Coroutine] = []
+        self.queue: list[list[Coroutine]] = []
         self.running = False
 
     @property
     def pending_tasks(self) -> int:
         """Return a count of pending tasks in the queue."""
-        return len(self.queue)
+        return sum(len(item) for item in self.queue)
 
     @property
     def has_pending_tasks(self) -> bool:
-        """Return a count of pending tasks in the queue."""
+        """Return if there are pending tasks in the queue."""
         return self.pending_tasks != 0
```

Comment:
```suggestion
        return len(self.queue) != 0
```
As we remove things from `queue` this should be a simple count instead of `pending_tasks` since that one now does a sum of all childs

Reply:
I think it's not obvious what is correct here. If removing the items early, we consider the tasks no longer in the queue when execution starts. If removing the items late, we consider the tasks no longer in the queue when execution is finished.